app/src/test/java/org/mozilla/fenix/settings/sitepermissions/SitePermissionsManageExceptionsPhoneFeatureFragmentTest.kt

```diff
@@ -0,0 +1,479 @@
+        )
+
+        var value = AutoplayValue.BlockAll(
+            label = "label",
+            rules = rules,
+            sitePermission = null
+        )
+
+        assertTrue(value.isSelected())
+
+        rules = rules.copy(
+            autoplayAudible = AutoplayAction.BLOCKED,
+            autoplayInaudible = AutoplayAction.ALLOWED
+        )
+
+        value = AutoplayValue.BlockAll(
+            label = "label",
+            rules = rules,
+            sitePermission = null
+        )
+
+        assertFalse(value.isSelected())
+
+        value = AutoplayValue.BlockAll(
+            label = "label",
+            rules = rules,
+            sitePermission = SitePermissions(
+                origin = "",
+                savedAt = 0L,
+                autoplayAudible = AutoplayStatus.BLOCKED,
+                autoplayInaudible = AutoplayStatus.BLOCKED
+            )
+        )
+
+        assertTrue(value.isSelected())
+
+        value = AutoplayValue.BlockAll(
+            label = "label",
+            rules = rules,
+            sitePermission = SitePermissions(
+                origin = "",
+                savedAt = 0L,
+                autoplayAudible = AutoplayStatus.ALLOWED,
+                autoplayInaudible = AutoplayStatus.BLOCKED
+            )
+        )
+
+        assertFalse(value.isSelected())
+    }
+
+    @Test
+    fun `BlockAudible - isSelected`() {
+        var rules = getRules().copy(
+            autoplayAudible = AutoplayAction.BLOCKED,
+            autoplayInaudible = AutoplayAction.ALLOWED
+        )
+
+        var value = AutoplayValue.BlockAudible(
+            label = "label",
+            rules = rules,
+            sitePermission = null
+        )
+
+        assertTrue(value.isSelected())
+
+        rules = rules.copy(
+            autoplayAudible = AutoplayAction.BLOCKED,
+            autoplayInaudible = AutoplayAction.BLOCKED
+        )
+
+        value = AutoplayValue.BlockAudible(
+            label = "label",
+            rules = rules,
+            sitePermission = null
+        )
+
+        assertFalse(value.isSelected())
+
+        value = AutoplayValue.BlockAudible(
+            label = "label",
+            rules = rules,
+            sitePermission = SitePermissions(
+                origin = "",
+                savedAt = 0L,
+                autoplayAudible = AutoplayStatus.BLOCKED,
+                autoplayInaudible = AutoplayStatus.ALLOWED
+            )
+        )
+
+        assertTrue(value.isSelected())
+
+        value = AutoplayValue.BlockAudible(
+            label = "label",
+            rules = rules,
+            sitePermission = SitePermissions(
+                origin = "",
+                savedAt = 0L,
+                autoplayAudible = AutoplayStatus.ALLOWED,
+                autoplayInaudible = AutoplayStatus.ALLOWED
+            )
+        )
+
+        assertFalse(value.isSelected())
+    }
+
+    @Test
+    fun `AllowAll - createSitePermissionsFromCustomRules`() {
```

Comment:
nit: I think all the test names in this class can be improved. Right now they just contain the method name. I personally really like the GIVEN/WHEN/THEN naming with given being optional. But if the tests fails it would be nice to know exactly hat assumption didn't hold, rather that just reading `createSitePermissionsFromCustomRules` failed.

Reply:
👍🏽

Reply:
I will update the tests names :)